packages/framework/src/Model/Payment/Payment.php

```diff
@@ -119,13 +120,31 @@ public function addTransport(Transport $transport)
     public function setTransports(array $transports)
     {
         $this->transports->clear();
+
         foreach ($transports as $transport) {
             $this->addTransport($transport);
         }
     }
 
     /**
-     * @return \Doctrine\Common\Collections\Collection
+     * @param \Shopsys\FrameworkBundle\Model\Transport\Transport $transport
+     */
+    public function removeTransport(Transport $transport)
+    {
+        $this->transports->removeElement($transport);
+        $transport->removePayment($this);
+    }
+
+    protected function clearTransports()
+    {
+        foreach ($this->transports as $transport) {
+            $this->removeTransport($transport);
+            $transport->removePayment($this);
```

Comment:
`removeTransport` calls also `$transport->removePayment($this)` This duplication will fail

Reply:
Fixed